fili-core/src/main/java/com/yahoo/bard/webservice/web/handlers/workflow/CacheMode.java

```diff
@@ -0,0 +1,108 @@
+// Copyright 2016 Yahoo Inc.
+// Licensed under the terms of the Apache license. Please see LICENSE.md file distributed with this work for terms.
+package com.yahoo.bard.webservice.web.handlers.workflow;
+
+import com.yahoo.bard.webservice.config.SystemConfig;
+import com.yahoo.bard.webservice.config.SystemConfigProvider;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Optional;
+
+/**
+ * Caching config analyzer.
+ */
+public final class CacheMode {
```

Comment:
I really wanted you to make CacheMode an interface and what you call Mode inside to be 'DefaultCacheMode'.